pages/am/3.x/user-guide/mfa/user-guide-mfa-security.adoc

```diff
@@ -0,0 +1,77 @@
+= MFA Security
+:page-sidebar: am_3_x_sidebar
+:page-permalink: am/current/am_userguide_mfa_security.html
+:page-folder: am/user-guide
+:page-layout: am
+
+== Overview
+
+[label label-version]#New in version 3.20#
+
+Gravitee access management introduces MFA rate limit  and brute force detection in `AM 3.20` release to enhance MFA security further.
+These security features are designed to make the multi-factor authentication process more resilient against a bad actor.
+These features allow:
+
+* specify the maximum number of MFA challenge the application can request to obtain SMS or Email verification code and
+* enforce the maximum attempts of MFA code verification
+
+== MFA Rate Limit
+
+Rate limit is a feature which helps to configure and limit the number of challenges a user is allowed to send in a specific time period.
+There are factors (such as SMS or email factor) which may incur cost due to the nature of the services or there could be a limited number of available requests in a certain amount of time.
+Now it is possible to address these circumstances by configuring and limiting the number of challenges a user can send to align with the underlying services.
+
+The rate limit configuration is available in the `gravitee.yaml` of the AM gateway under `mfa_rate` section. The configuration is disabled by default.
+The code fragment below is an enabled sample configuration which is self-explanatory. The `timeUnit` value could be `Hours`, `Minutes` or `Seconds`.
+
+[source, yaml]
+----
+mfa_rate:
+  enabled: true
+  limit: 5
+  timePeriod: 15
+  timeUnit: Minutes
+----
+
+You can define these properties in the `values.yaml` file of the AM Helm Chart as well.
+
+[source, yaml]
+----
+gateway:
+  mfa_rate:
+    enabled: true
+    limit: 5
+    timePeriod: 15
+    timeUnit: Minutes
+----
+
+Gravitee AM monitors the MFA challenge request based on the enabled rate limit configuration and shows the user rate limit exceeded message if the user exceeds the limit.
+Then the user has to wait a certain amount of time before making a successful request.
+For an example; if the limit is set to 2 for 1 minute time period and the user already has sent 2 requests, then the user must wait another 30 seconds before being able to send the 3rd request.
+Here is a screenshot of the challenge step with exhausted rate limit:
+
+image::{% link images/am/current/graviteeio-am-userguide-mfa-ratelimit-exceed.png %}[250, 500]
+
+NOTE: You can customise the error message by setting `mfa_challenge.rate.limit.error` property value in the *messages_en.properties* or *messages_fr.properties* file.
+
+
+== Brute Force detection
+
+Brute force is a feature which helps to configure and limit the number of verification requests a user is allowed to send in a specific time period.
+This feature is configurable in domain or in application level. Follow the steps below to configure the *Brute Force* feature in domain level
```

Comment:
```suggestion
You can configure Brute Force Detection at domain level or at application level. Follow the steps below to configure the *Brute Force* feature at domain level.
```

Reply:
yes, exactly the same screen at application level than at domain level.